src/Components/Server/src/Circuits/CircuitPrerenderer.cs

```diff
@@ -40,6 +45,13 @@ public async Task<IEnumerable<string>> PrerenderComponentAsync(ComponentPrerende
             }
         }
 
+        private void CircuitHost_UnhandledException(object sender, UnhandledExceptionEventArgs e)
+        {
+            // Throw all exceptions encountered during pre-rendering so the default developer
+            // error page can respond.
+            throw (Exception)e.ExceptionObject;
```

Comment:
Shouldn't this be ExceptionDispatchInfo.Capture(e).Throw() to preserve the original callstack?

Reply:
👍